fastflix/widgets/panels/status_panel.py

```diff
@@ -111,6 +121,40 @@ def update_bitrate(self, bitrate):
     def update_title_bar(self):
         pass
 
+    def set_started_at(self, msg):
+        try:
+            started_at = datetime.datetime.fromisoformat(msg.split("__")[-1])
+        except Exception as e:
+            logger.warning(f"Unable to parse start time - {str(e)}")
+            return
+
+        self.started_at = started_at
+
+    def update_time_elapsed(self):
+        now = datetime.datetime.now(datetime.timezone.utc)
+
+        if not self.started_at:
+            logger.warning("Unable to update time elapsed because start time isn't set")
+            return
+
+        try:
+            time_elapsed = now - self.started_at
+        except Exception as e:
+            logger.warning(f"Unable to calculate elapsed time - {str(e)}")
+            return
+
+        self.time_elapsed_label.setText(f"{t('Time Elapsed')}: {timedelta_to_str(time_elapsed)}")
+
+    def on_status_update(self, msg):
+        update_type = msg.split("__")[0]
+
+        if update_type in ["complete", "cancelled", "error"]:
+            self.time_elapsed_ticker_thread.running = False
```

Comment:
This is where we would want to use a new signal to control the thread instead of changing it's properties directly. Best to have any iterations between a thread and main process done via a signal  (but maybe best to just terminate thread when done, see next comment)

Reply:
That makes very much sense! Attempt added in c1e86d2